### PR TITLE
fix(build): Removed experimental flags for MSVC

### PR DESCRIPTION
**Pull request by capnkenny**
Removed `/experimental:external` flag as it's no longer experimental as of VS 2019 16.10.
Made `BUILD_INTERFACE` genex evaluate first when adding external headers for third-party dependencies.

### CODE REVIEW

File: thirdparty/json/CMakeLists.txt
Comment (RubyNova) [inline thirdparty/json/CMakeLists.txt]: We're missing EOF here. I know that wasn't introduced by this change but we should fix it.

File: CMakeLists.txt
Comment (RubyNova) [inline CMakeLists.txt:5]: What's this doing?

**Review by RubyNova — CHANGES_REQUESTED** [review]
A question and a file formatting error that wasn't introduced by this PR should probably be fixed while we are here.

Otherwise LGTM assuming a test in the NovelRT CMake proves this works.

File: CMakeLists.txt
Comment (FiniteReality) [reply CMakeLists.txt:5]: Apparently it sets the extraction time to the time of extraction, rather than what is specified in the archive. I'm guessing this is because of caching or something.

Comment (capnkenny) [reply CMakeLists.txt:5]: Yes - CMake 3.24 introduce a new policy that affects all FetchContent calls that utilize `ExternalProject_Add`:
> When using the URL download method with the [ExternalProject_Add()](https://cmake.org/cmake/help/latest/module/ExternalProject.html#command:externalproject_add) command, CMake 3.23 and below sets the timestamps of the extracted contents to the same as the timestamps in the archive. When the URL changes, the new archive is downloaded and extracted, but the timestamps of the extracted contents might not be newer than the previous contents. Anything that depends on the extracted contents might not be rebuilt, even though the contents may change.
> CMake 3.24 and above prefers to set the timestamps of all extracted contents to the time of the extraction. This ensures that anything that depends on the extracted contents will be rebuilt whenever the URL changes.

> The DOWNLOAD_EXTRACT_TIMESTAMP option to the [ExternalProject_Add()](https://cmake.org/cmake/help/latest/module/ExternalProject.html#command:externalproject_add) command can be used to explicitly specify how timestamps should be handled. When DOWNLOAD_EXTRACT_TIMESTAMP is not given, this policy controls the default behavior. The OLD behavior for this policy is to restore the timestamps from the archive. The NEW behavior sets the timestamps of extracted contents to the time of extraction.

File: thirdparty/json/CMakeLists.txt
Comment (capnkenny) [reply thirdparty/json/CMakeLists.txt]: No idea where it is lol - time to find it cuz the preview doesn't show it :^)

File: CMakeLists.txt
Comment (capnkenny) [reply CMakeLists.txt:5]: TL;DR - CMake 3.24 makes sure that downloaded items get rebuilt if the URL changes based on timestamp - doesn't affect lower versions as the policy doesn't exist, and w/o setting this the configuration fails in CI.

File: thirdparty/json/CMakeLists.txt
Comment (RubyNova) [reply thirdparty/json/CMakeLists.txt]: image

**Review by RubyNova — APPROVED** [review]
LGTM pending internal NovelRT test. Please make sure it works there before merging.

**Comment by capnkenny** [comment]
This has passed testing and works when referenced properly. Merging.

**Pull request merged**
